commit f5d834fc34e61f1a40435981062000e5d2b2baa8
tree 3ed0b4a2ba4870b733e66483b01d69903142aa59
parent 16782a604c458e1edcefca52457a82395b788bed
author Alan Curry <pacman@theworld.com> 1185326912 +1000
committer Paul Mackerras <paulus@samba.org> 1185430665 +1000

    [POWERPC] Fix Pegasos keyboard detection
    
    As of 2.6.22 the kernel doesn't recognize the i8042 keyboard/mouse
    controller on the PegasosPPC.  This is because of a feature/bug in the
    OF device tree: the "device_type" attribute is an empty string instead
    of "8042" as the kernel expects.  This adds a secondary detection
    which looks for a device whose *name* is "8042" if there is no device
    whose *type* is "8042".
    
    Signed-off-by: Alan Curry <pacman@world.std.com>
    Acked-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
    Signed-off-by: Paul Mackerras <paulus@samba.org>

---
 arch/powerpc/kernel/setup-common.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/powerpc/kernel/setup-common.c	2007-07-08 19:32:17.000000000 -0400
+++ b/arch/powerpc/kernel/setup-common.c	2007-08-27 14:01:19.000000000 -0400
@@ -487,6 +487,10 @@ int check_legacy_ioport(unsigned long ba
 	switch(base_port) {
 	case I8042_DATA_REG:
 		np = of_find_node_by_type(NULL, "8042");
+		/* Pegasos has no device_type on its 8042 node, look for the
+		 * name instead */
+		if (!np)
+			np = of_find_node_by_name(NULL, "8042");
 		break;
 	case FDC_BASE: /* FDC1 */
 		np = of_find_node_by_type(NULL, "fdc");
